src/components/DataCollector/index.js

```diff
@@ -63,8 +63,12 @@ const createDataCollector = ({ config, logger }) => {
         // it overlays on top of any data Alloy automatically
         // provides. This allows the user to override the
         // automatically collected data.
-        event.mergeXdm(xdm);
-        event.data = data;
+        if (xdm) {
+          event.mergeXdm(xdm);
+        }
+        if (data) {
+          event.data = data;
+        }
```

Comment:
I think it's probably a good idea to only set these properties on `event` if we actually have values for them, but it's also important for this temporary fix to work, since we would be overwriting the default `data` object with `undefined` if the customer didn't pass in `data` as an option.